x-pack/plugins/infra/public/alerting/log_threshold/components/expression_editor/editor.tsx

```diff
@@ -299,6 +306,21 @@ export const Editor: React.FC<
 
       {alertParams.criteria && isRatioAlert(alertParams.criteria) && criteriaComponent}
 
+      {shouldShowGroupByOptimizationWarning && (
+        <>
+          <EuiSpacer size="l" />
+          <EuiCallOut color="warning">
+            {i18n.translate('xpack.infra.logs.alertFlyout.groupByOptimizationWarning', {
+              defaultMessage:
+                'When setting a group by we highly recommend using the {comparator} comparator for your threshold. This can lead to significant performance improvements.',
```

Comment:
Open to thoughts on the wording here, I was trying to convey the problem without getting into implementation details / things that the user shouldn't need to know, whilst still trying to nudge in the right direction.

Reply:
Yeah, that does seem to make things a bit easier to parse.